### PR TITLE
Add non-generic Find to DbContext

### PR DESCRIPTION
**Pull request by ajcvickers**
Issue #5998

Moved Find code into its own service and re-factored so it could also be used in a non-generic way.

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore/Internal/IEntityFinder.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        object Find([NotNull] params object[] keyValues);
+
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        Task<object> FindAsync([NotNull] params object[] keyValues);
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore/Internal/IEntityFinder.cs]: Remove this and use a default value for cancellationToken.

File: src/Microsoft.EntityFrameworkCore/Internal/IEntityFinder`.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        new TEntity Find([NotNull] params object[] keyValues);
+
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        new Task<TEntity> FindAsync([NotNull] params object[] keyValues);
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore/Internal/IEntityFinder`.cs]: Remove this and use a default value for cancellationToken.

**Comment by anpete** [comment]
:shipit:

File: src/Microsoft.EntityFrameworkCore/Internal/IEntityFinder`.cs (hunk 2 above)
Comment (ajcvickers) [inline src/Microsoft.EntityFrameworkCore/Internal/IEntityFinder`.cs]: Presumably you mean by making the API non-params?

**Pull request merged**